core/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -315,6 +315,25 @@ protected BytesRef indexedValueForSearch(Object value) {
         return BytesRefs.toBytesRef(value);
     }
 
+    /** Returns true if the field is searchable.
+     *
+     */
+    public boolean isSearchable() {
+        return indexOptions() != IndexOptions.NONE;
+    }
```

Comment:
Unfortunately I can't, this is used by the inner NumberType in NumberFieldMapper.